server/game/gamesteps/triggeredabilitywindow.js

```diff
@@ -95,6 +95,9 @@ class TriggeredAbilityWindow extends BaseAbilityWindow {
             if(abilityChoice.text !== 'default') {
                 title += ' - ' + abilityChoice.text;
             }
+            if(abilityChoice.card.location !== 'hand' && abilityChoice.card.location !== 'play area') {
```

Comment:
Per the test failures, you also need to allow it to be triggered from `'active plot'` and `'agenda'` locations without tacking on the location.

For Freedmen and things like The Prince's Plan that normally trigger from discard, you can probably boil this check down entirely to:
```javascript
if(abilityChoice.card.location !== abilityChoice.ability.location) {
```

which should only pick up abnormal cases (e.g. Annals event from discard)